R/data_separate.R

```diff
@@ -0,0 +1,403 @@
+        "max" = max(l, na.rm = TRUE),
+        "mode" = distribution_mode(l),
+      )
+      # tell user
+      if (verbose && !all(l == 1) && !is.numeric(separator)) {
+        insight::format_alert(paste0(
+          "Column `", sep_column, "` had different number of values after splitting. Variable was split into ",
+          n_cols, " column", ifelse(n_cols > 1, "s", ""), "."
+        ))
+      }
+    } else {
+      # else, if we know number of columns, use that number
+      n_cols <- n_columns
+    }
+
+    # main task here - fill or drop values for all columns
+    separated_columns <- tryCatch(
+      .fix_separated_columns(separated_columns, fill, extra, n_cols, sep_column, verbose),
+      error = function(e) NULL
+    )
+
+    # catch error
+    if (is.null(separated_columns)) {
+      insight::format_error(
+        "Something went wrong. Probably the number of provided column names did not match number of newly created columns?"
+      )
+    }
+
+    # bind separated columns into data frame and set column names
+    out <- as.data.frame(do.call(rbind, separated_columns))
+
+    # if no column names provided, use standard names
+    if (is.null(new_columns[[sep_column]])) {
+      new_column_names <- paste0(sep_column, "_", seq_along(out))
+    } else {
+      # if we have multiple columns that were separated, we avoid duplicated
+      # column names of created variables by appending name of original column
+      if (make_unique_colnames) {
+        new_column_names <- paste0(sep_column, "_", new_columns[[sep_column]])
+      } else {
+        new_column_names <- new_columns[[sep_column]]
+      }
+    }
+
+    colnames(out) <- new_column_names
+    out
+  })
+
+  # any split performed?
+  if (all(lengths(split_data) == 1)) {
+    if (verbose) {
+      insight::format_alert("Separator probably not found. No values were split. Returning original data.")
+    }
+    return(data)
+  }
+
+  # final preparation, bind or merge columns, make unique columm names
+  if (isTRUE(merge_multiple) && length(split_data) > 1) {
+    # we merge all split columns, which are currently saved as list
+    # of data frames, together into one data frame
+    for (i in 2:length(split_data)) {
+      for (j in seq_along(split_data[[1]])) {
+        split_data[[1]][[j]] <- gsub(" ", "",
+          paste(
+            split_data[[1]][[j]],
+            split_data[[i]][[j]],
+            sep = merge_separator
+          ),
+          fixed = TRUE
+        )
+      }
+    }
+    split_data <- split_data[[1]]
+  } else {
+    # bind all columns
+    split_data <- do.call(cbind, split_data)
+    colnames(split_data) <- make.unique(colnames(split_data))
+  }
+
+  # convert "NA" strings into real NA?
+  if (convert_na) {
+    split_data[] <- lapply(split_data, function(i) {
+      i[i == "NA"] <- NA_character_
+      i
+    })
+  }
+
+  if (isTRUE(append)) {
+    data <- cbind(data, split_data)
+  } else {
+    data <- split_data
+  }
```

Comment:
```suggestion
data <- cbind(data, split_data)
if (!isTRUE(append)) {
   data[select] <- NULL
}
```
we don't want to drop the whole dataset (same as in `data_unite()`)